packages/calling-component-bindings/src/handlers/createCommonHandlers.ts

```diff
@@ -57,16 +53,32 @@ export type CallingHandlersCommon = {
   onDisposeLocalStreamView: () => Promise<void>;
   /* @conditional-compile-remove(dialpad) */ /* @conditional-compile-remove(PSTN-calls) */
   onSendDtmfTone: (dtmfTone: DtmfTone) => Promise<void>;
-  onStartCall: (
-    participants: (CommunicationUserIdentifier | PhoneNumberIdentifier | UnknownIdentifier)[],
-    options?: StartCallOptions
-  ) => void;
   onRemoveParticipant(userId: string): Promise<void>;
   /* @conditional-compile-remove(PSTN-calls) */
   onRemoveParticipant(participant: CommunicationIdentifier): Promise<void>;
   /* @conditional-compile-remove(call-readiness) */
   askDevicePermission: (constrain: PermissionConstraints) => Promise<void>;
 };
+
+/**
+ * Object containing all the handlers required for calling components.
+ *
+ * Calling related components from this package are able to pick out relevant handlers from this object.
+ * See {@link useHandlers} and {@link usePropsFor}.
+ *
+ * @public
+ */
+// CommonCallingHandlers is an interface compatible with TeamsCallingHandlers and CallingHandlers
+// which is also a common interface required by components and adapters
+// components and adapters only requires onStartCall is a function returning void
+// but in TeamsCallingHandlers and CallingHandlers, it returns different types for non-breaking change reason
+export type CommonCallingHandlers = SharedCallingHandlers & {
```

Comment:
This is unfortunate, and I'd guess mighty confusing for Contoso.

does this work?

```

interface SharedCallingHandlers {
  onStartCall(...args): void;
}

interface CallingHandlers {
  onStartCall(...args): Call;
}

interface TeamsCallingHandlers {
  onStartCall(...args): TeamsCall;
}
```

i.e., the common interface has a method that returns `void` and the specific interfaces extend that method with different return types.

Reply:
> mighty confusing for Contoso

I would add and us developing in this repo. `CommonCallingHandlers` and `SharedCallingHandlers` by name are essentially synonyms

Reply:
@prprabhu-ms That was the first shot I tried, but it does not allow me to override with same parameter but different return value, then I chose this one in the end

@prprabhu-ms @JamesBurnside Do you have a better naming recommendation?

Reply:
What is the error you got? Typescript seems to be happy with:

```Typescript

interface A {
  foo(x: number): void;
}

interface B extends A {
  foo(x: number): number;
}

interface C extends A {
  foo(x: number): string;
}

function main() {
  const b: B = {
    foo: (x: number) => { return x; }
  }
  iWantA(b);
  const c: C = {
    foo: (x: number) => { return `${x}`; }
  }
  iWantA(c);
}

function iWantA(a: A) {
  a.foo(2);
}
```

Reply:
Interesting, I tried Interface, and it does override the function, 

![image](https://user-images.githubusercontent.com/11863655/201242809-9bd8130b-aaa5-4ccb-b845-0f271b587b31.png)

but it does not work for Type:
![image](https://user-images.githubusercontent.com/11863655/201242884-f29190af-726e-4993-8627-e4729b1a93d7.png)


Then there is one more obvious reason when we want to use Interface over Type - for better function override. Lemme change all the definition to Interface for this case and see if it works well

Reply:
This is an awesome discussion! Learning something new here